src/scanner.c

```diff
@@ -382,6 +385,7 @@ UInt OpenInputStream (
     }
 
     /* enter the file identifier and the file name                         */
+    assert(TLS(Input) != 0);
```

Comment:
I was confused by your remark at first, but unfolding the diff above, I see there is an `if` above to which this applies... however, it seems github forces comments on that line to move down here...